File: Going Dutch/Views/iPhone/MCPaymentPresenceTableViewCell_iPhone.m
//
//  MCPaymentPresenceTableViewCell_iPhone.m
//  We all pay
//
//  Created by Mark Cornelisse on 15-05-14.
//  Copyright (c) 2014 Mark Cornelisse. All rights reserved.
//

#import "MCPaymentPresenceTableViewCell_iPhone.h"

#import "MCPaymentPresence+addons.h"
#import "MCPayment+addons.h"

@implementation MCPaymentPresenceTableViewCell_iPhone

#pragma mark - Actions

- (IBAction)presenceIsSwitched:(id)sender
{
    [_thisCellsPaymentPresence setIsPersonPresent:@([_isPresentSwitch isOn])];
    [[_thisCellsPaymentPresence payment] recalculateAveragePeopleOweAndStore];
}

#pragma mark - New in this class

#pragma mark - Inherited from super

- (id)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
{
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        // Initialization code
    }
    return self;
}

- (void)awakeFromNib
{
    // Initialization code
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated
{
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

@end

File: Going Dutch/MCPaymentPresenceTableViewCell_iPhone.swift
//
//  MCPaymentPresenceTableViewCell_iPhone.swift
//  We all pay
//
//  Created by Mark Cornelisse on 11/06/15.
//  Copyright (c) 2015 Mark Cornelisse. All rights reserved.
//

import UIKit

import FirebaseAnalytics

class MCPaymentPresenceTableViewCell_iPhone: UITableViewCell {
    @IBOutlet var personView: UIImageView!
    @IBOutlet var nameLabel: UILabel!
    @IBOutlet var isPresentSwitch: UISwitch!
    @IBOutlet var owesLabel: UILabel!
    @objc var thisCellsPaymentPresence: MCPaymentPresence?
    
    @IBAction func presenceIsSwitched(_ sender: UISwitch) {
        Analytics.logEvent("SwitchPresence on payment", parameters: ["Presence Value": (isPresentSwitch.isOn as NSNumber)])
        thisCellsPaymentPresence!.isPersonPresent = NSNumber(value: isPresentSwitch.isOn)
        thisCellsPaymentPresence!.payment.recalculateAveragePeopleOweAndStore()
    }
}
